internal/server/proto/server.proto

```diff
@@ -319,8 +319,78 @@ service Waypoint {
 
   // UpsertStatusReport updates or inserts a statusreport.
   rpc UpsertStatusReport(UpsertStatusReportRequest) returns (UpsertStatusReportResponse);
+
+  //----------------------------------------------------------------------
+  // UI endpoints. The endpoints below are convenience methods for waypoint's
+  // various UIs (web GUI and CLI). They contain no unique functionality, i.e.
+  // anything achievable through these endpoints should also be achievable through
+  // a combination of calls to non-UI endpoints.
+  //----------------------------------------------------------------------
+
+  // List deployments for a given application.
+  rpc UI_ListDeployments(UI.ListDeploymentsRequest) returns (UI.ListDeploymentsResponse);
+
+  // List releases for a given application.
+  rpc UI_ListReleases(UI.ListReleasesRequest) returns (UI.ListReleasesResponse);
+}
+
+/********************************************************************
+* UI
+********************************************************************/
+message UI {
+
+  message ListDeploymentsRequest {
+    // Application that the deployments belong to.
+    Ref.Application application = 1;
+
+    // Workspace that the deployments belong to.
+    // If undefined, deployments from all workspaces will be listed.
+    Ref.Workspace workspace = 2;
+
+    OperationOrder order = 3;
```

Comment:
Looks like the RFC had pagination settings, we should include that I think.

Reply:
I left it out for now because I don't plan to implement it right now, and I thought it would be dishonest to have the field that doesn't do anything.

Reply:
Good thing you didn't touch the pagination stuff because [further input on the RFC](https://docs.google.com/document/d/1RHuMDPJlWPWoZYx9AkRTw8ppHkg9j7RZwBjA4uUcbcQ/edit?disco=AAAANZQwK30) from @dadgar suggests we should revisit the pagination schema.

I think I’ll write a short follow-up RFC, but it’ll probably be a little simpler/flatter:

```diff
  message ListDeploymentsRequest {
    // Application that the deployments belong to.
    Ref.Application application = 1;

    // Workspace that the deployments belong to.
    // If undefined, deployments from all workspaces will be listed.
    Ref.Workspace workspace = 2;

-   OperationOrder order = 3;
+   OperationOrder deprecated_order = 3;

    // The filters to apply to this request. These are ORed, so you should
    // specify multiple filters in the StatusFilter for AND behavior.
    repeated StatusFilter status = 4;

    // The physical state to filter for. If this is zero or unset then no
    // filtering on physical state will be done.
    Operation.PhysicalState physical_state = 5;

+   repeated string order_by = 6;
+   uint32 page_size = 7;
+   string next_page_token = 8;
+   string previous_page_token = 9;
  }
```